assets/js/edit-task.js

```diff
@@ -0,0 +1,37 @@
+import { CLICK_EVENT } from "/assets/js/constants.js";
+
+function editTaskEvent(done, edit, dlt, revert, saveButton, inputElement, tools, allTasks, id) {
+  edit.addEventListener(CLICK_EVENT, () => {
+    const prevContent = inputElement.innerText;
+    inputElement.contentEditable = true;
+    inputElement.focus();
+    edit.classList.add("hide");
+    dlt.classList.add("hide");
+    revert.classList.remove("hide");
+    saveButton.classList.remove("hide");
+
+    function saveEvent() {
+      inputElement.contentEditable = false;
+      saveButton.classList.add("hide");
+      dlt.classList.remove("hide");
+      revert.classList.add("hide");
+      allTasks[id].content = inputElement.textContent;
+    }
+
+    saveButton.addEventListener(CLICK_EVENT, () => {
+      saveEvent();
+      edit.classList.remove("hide");
+    });
+
+    done.addEventListener(CLICK_EVENT, () => {
+      saveEvent();
+    });
+
+    revert.addEventListener(CLICK_EVENT, (event) => {
+      inputElement.innerText = prevContent;
+      saveEvent();
+      edit.classList.remove("hide");
+    });
```

Comment:
This part of the code is for making the task state as done on edit, right?
Can we write a separate function for that and call that instead?